modules/nvidia_plugin/src/transformer/detection_output_fix_input_types_transformation.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "detection_output_fix_input_types_transformation.hpp"
+
+#include "openvino/cc/pass/itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/detection_output.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace ov::pass::pattern;
+
+namespace ov::nvidia_gpu::pass {
+
+bool detection_output_fix_input_types(Matcher& m) {
+    const auto d_out = std::dynamic_pointer_cast<ov::op::v0::DetectionOutput>(m.get_match_root());
+    if (!d_out) {
+        return false;
+    }
+
+    const auto& type = d_out->get_element_type();
+
+    const auto& inputs = d_out->inputs();
+    ov::NodeVector input_nodes;
+    bool needs_transform = false;
+    for (std::size_t i = 0; i < d_out->inputs().size(); ++i) {
+        const auto& input_type = d_out->input(i).get_element_type();
+        if (input_type != type) {
+            needs_transform = true;
+            const auto convert = std::make_shared<ov::op::v0::Convert>(d_out->input_value(i), type);
+            ov::copy_runtime_info(d_out, convert);
+            input_nodes.emplace_back(convert);
+        } else {
+            input_nodes.emplace_back(d_out->input_value(i).get_node_shared_ptr());
+        }
+    }
+    if (!needs_transform) {
+        return false;
+    }
+
+    const auto inputs_size = input_nodes.size();
+    const auto& attrs = d_out->get_attrs();
+    std::shared_ptr<ov::op::v0::DetectionOutput> new_d_out{nullptr};
+    if (inputs_size == 3) {
+        new_d_out =
+            std::make_shared<ov::op::v0::DetectionOutput>(input_nodes[0], input_nodes[1], input_nodes[2], attrs);
+    } else if (inputs_size == 5) {
+        new_d_out = std::make_shared<ov::op::v0::DetectionOutput>(
+            input_nodes[0], input_nodes[1], input_nodes[2], input_nodes[3], input_nodes[4], attrs);
+    } else {
+        return false;
+    }
+    new_d_out->set_friendly_name(d_out->get_friendly_name());
+    ov::copy_runtime_info(d_out, new_d_out);
+    ov::replace_node(d_out, new_d_out);
+
+    return true;
+}
+
+DetectionOutputFixInputTypesTransformation3::DetectionOutputFixInputTypesTransformation3() {
+    MATCHER_SCOPE(DetectionOutputFixInputTypesTransformation3);
+
+    const auto d_out = wrap_type<ov::op::v0::DetectionOutput>({any_input(), any_input(), any_input()});
+    matcher_pass_callback callback = [](Matcher& m) { return detection_output_fix_input_types(m); };
+
+    const auto m = std::make_shared<Matcher>(d_out, matcher_name);
+    register_matcher(m, callback);
+}
+
+DetectionOutputFixInputTypesTransformation5::DetectionOutputFixInputTypesTransformation5() {
+    MATCHER_SCOPE(DetectionOutputFixInputTypesTransformation5);
+
+    const auto d_out =
+        wrap_type<ov::op::v0::DetectionOutput>({any_input(), any_input(), any_input(), any_input(), any_input()});
```

Comment:
Why do we need 2 similiar matchers? Why not use just `wrap_type<ov::op::v0::DetectionOutput>();`?

Reply:
Done